Source/Scene/TileBoundingS2Cell.js

```diff
@@ -0,0 +1,695 @@
+ * ---\---------/---   ---\---------/---   ---X---------/---   ---\---------/---
+ *     \       /           \       /           \       /           \       /
+ *   ---\-----/---       ---\-----/---       ---\-----/---       ---\-----/---
+ *       \   /               \   /               \   /               \   /
+ *                                                                    \ /
+ *                                                                     \
+ *                                                                    / \
+ *                                                                   / X \
+ *
+ *         I                  II                  III                 IV
+ *
+ * Case I: There is only one plane selected.
+ * In this case, we project the point onto the plane and do a point polygon distance check to find the closest point on the polygon.
+ * The point may lie inside the "face" of the polygon or outside. If it is outside, we need to determine which edges to test against.
+ *
+ * Case II: There are two planes selected.
+ * In this case, the point will lie somewhere on the line created at the intersection of the selected planes.
+ *
+ * Case III: There are three planes selected.
+ * In this case, the point will lie on the vertex, at the intersection of the selected planes.
+ *
+ * Case IV: There are more than three planes selected.
+ * Since we are on an ellipsoid, this will only happen in the bottom plane, which is what we will use for the distance test.
+ */
+TileBoundingS2Cell.prototype.distanceToCamera = function (frameState) {
+  //>>includeStart('debug', pragmas.debug);
+  Check.defined("frameState", frameState);
+  //>>includeEnd('debug');
+
+  var point = frameState.camera.positionWC;
+
+  var selectedPlaneIndices = [];
+  var vertices;
+  var edgeNormals;
+  var rotation;
+
+  if (
+    CesiumMath.greaterThan(
+      Plane.getPointDistance(this._boundingPlanes[0], point),
+      0,
+      CesiumMath.EPSILON14
+    )
+  ) {
+    selectedPlaneIndices.push(0);
+    vertices = this._vertices.slice(0, 4);
+    edgeNormals = this._edgeNormals[0];
+    rotation = -1;
+  } else if (
+    CesiumMath.greaterThan(
+      Plane.getPointDistance(this._boundingPlanes[1], point),
+      0,
+      CesiumMath.EPSILON14
+    )
+  ) {
+    selectedPlaneIndices.push(1);
+    vertices = this._vertices.slice(4, 8);
+    edgeNormals = this._edgeNormals[1];
+    rotation = 1;
+  }
+
+  var i;
+  var sidePlaneIndex;
+  for (i = 0; i < 4; i++) {
+    sidePlaneIndex = 2 + i;
+    if (
+      CesiumMath.greaterThan(
+        Plane.getPointDistance(this._boundingPlanes[sidePlaneIndex], point),
+        0,
+        CesiumMath.EPSILON14
+      )
+    ) {
+      selectedPlaneIndices.push(2 + i);
+      vertices = [
+        this._vertices[i % 4],
+        this._vertices[4 + i],
+        this._vertices[4 + ((i + 1) % 4)],
+        this._vertices[(i + 1) % 4],
+      ];
+      edgeNormals = this._edgeNormals[2 + i];
+      rotation = -1;
```

Comment:
I see that these variables are clobbered if the point is between two side planes (e.g. near the edge) -- I presume that's intended? Also it seems that means the code is assuming a CCW iteration order. Might want to comment that the iteration order matters if so.